src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -324,6 +324,10 @@ public File getRootDirFor(I child) {
         return getRootDirFor(child.getName()); // TODO see comment regarding loadChildren and encoding
     }
 
+    /**
+     * It is unwise to override this, lest links to children from nondefault {@link View}s break.
```

Comment:
Of course if the views were written correctly in the first place then it would be ok to override this